pkg/query/query.go

```diff
@@ -143,16 +144,27 @@ func (q *Query) Query(ctx context.Context, req *pb.QueryRequest) (*pb.QueryRespo
 		return nil, status.Error(codes.InvalidArgument, err.Error())
 	}
 
+	var (
+		p   storage.InstantProfile
+		err error
+	)
+
 	switch req.Mode {
 	case pb.QueryRequest_MODE_SINGLE_UNSPECIFIED:
-		return q.singleRequest(ctx, req.GetSingle())
+		p, err = q.selectSingle(ctx, req.GetSingle())
 	case pb.QueryRequest_MODE_MERGE:
-		return q.mergeRequest(ctx, req.GetMerge())
+		p, err = q.mergeRequest(ctx, req.GetMerge())
 	case pb.QueryRequest_MODE_DIFF:
-		return q.diffRequest(ctx, req.GetDiff())
```

Comment:
Rendering was intentionally done in these methods because for example flamegraphs need different data when diffing compared to a pprof export (which only contains one value, the diff, while diff flamegraphs consist of the "second" selection and are colored using the diff values).

The decision we were faced with was to either have these functions take care of rendering as well or have singleRequest and mergeRequest return nil "diff Profiles" to be passed to the flamegraph.